### PR TITLE
explicitly default port mapping respecting IP family

### PR DESCRIPTION
**Pull request by BenTheElder**
/cc @amwat @aojea 
fixes https://github.com/kubernetes-sigs/kind/issues/1292

### CODE REVIEW

File: pkg/cluster/internal/providers/docker/provision.go [hunk 1]
```diff
@@ -272,23 +274,39 @@ func generateMountBindings(mounts ...config.Mount) []string {
...
-		if pm.ListenAddress != "" {
-			hostPortBinding = net.JoinHostPort(pm.ListenAddress, fmt.Sprintf("%d", pm.HostPort))
-		} else {
-			hostPortBinding = fmt.Sprintf("%d", pm.HostPort)
+		// do provider internal defaulting
+		// in a future API revision we will handle this at the API level and remove this
+		if pm.ListenAddress == "" {
+			switch clusterIPFamily {
+			case config.IPv4Family:
+				pm.ListenAddress = "0.0.0.0" // this is the docker default anyhow
```
Comment (aojea) [inline pkg/cluster/internal/providers/docker/provision.go]: @liggitt enlightened me about these unspecified addresses, it turns out the go library behaves the same with "0.0.0.0" and "::"

Comment (BenTheElder) [reply pkg/cluster/internal/providers/docker/provision.go]: dockerd doesn't though, it uses a proxy process for `::` and iptables for `0.0.0.0`

**Comment by BenTheElder** [comment]
fixed typo in string unkown -> unknown

File: pkg/cluster/internal/providers/docker/provision.go [hunk 2]
```diff
@@ -272,23 +274,39 @@ func generateMountBindings(mounts ...config.Mount) []string {
...
+			case config.IPv6Family:
```
Comment (aojea) [inline pkg/cluster/internal/providers/docker/provision.go]: why not checking only for IPv6 now and keep defaulting as before?

**Comment by aojea** [comment]
can't we default the pm.	ListenAddress of portmaps in the API validation stage?

**Comment by BenTheElder** [comment]
Please read the code comments 🙃

In the future we should rethink API defaulting but right now that's left up to the provider.

File: pkg/cluster/internal/providers/docker/provision.go (hunk 2 above)
Comment (BenTheElder) [reply pkg/cluster/internal/providers/docker/provision.go]: Because the same problem will happen in dual stack ...

**Comment by aojea** [comment]
ahhg, it's too late for me, will lgtm tomorrow morning if nobody does it before 💤

**Pull request merged**